File: server/BeerWorkshop.Application/Dto/Requests/ProductsInventory/UpdateProductInventoryItemRequestDto.cs
using BeerWorkshop.Database.Enums;

namespace BeerWorkshop.Application.Dto.Requests.ProductsInventory;

public record UpdateProductInventoryItemRequestDto(
    decimal Quantity,
    DateTime IncomingDate,
    DateTime? OpeningDate,
    decimal PurchasePrice,
    decimal PurchaseVat,
    decimal RetailPrice,
    decimal PricePerQuantity,
    DateTime ManufactureDate,
    int ExpirationTime,
    ExpirationMeasure ExpirationMeasure,
    ExpirationCountingDateType ExpirationCountingDateType
);

File: client/src/redux/dtos/requests/productsInventory/updateProductInventoryItemRequestDto.ts
import {
  ExpirationCountingDateType,
  ExpirationMeasureType,
} from "../../../enums";

export interface UpdateProductInventoryItemRequestDto {
  quantity: number;
  incomingDate: Date;
  purchasePrice: number;
  retailPrice: number;
  pricePerQuantity: number;
  manufactureDate: Date;
  expirationTime: number;
  expirationMeasure: ExpirationMeasureType;
  openingDate?: Date | null;
  expirationCountingDateType: ExpirationCountingDateType;
}
